test/customer/postCustomerLoginIT.js

```diff
@@ -147,6 +147,45 @@ describe('magento postCustomerLogin', function() {
                 });
         });
 
+        //this test is based on the previous test which removes all entries from a customer cart
+        //it might fail if the customer cart is not empty
+        it('successfully login a customer & merge a cart', function() {
```

Comment:
I would prefer not. I started to design such test where I can check the merging result for different scenarios but IMO that is magento code and it doesn't add any value while it complicates our testing.

Reply:
This is an integration test here, so it is supposed to cover both our connector code and Magento. And I do see additional value, since we make sure that multiple cases in the "cart merge" scenario work using our API.

Reply:
This is out of context IMO. My understanding was that we should test a successful case scenario and all possible error cases or to simplify we have to test the API. Where do we stop if we start this, how do we track it and what about the existing cases? What about commerce tools?

Reply:
I don't have a strong opinion on this. Our ITs are meant to test use cases on our API independently of the underlying commerce engine. So we have to decide whether or not merging an existing customer cart with an anonymous cart after a user login is a valid use case of our API or not. I leave the decision up to you.

Reply:
But that's already tested, merging a customer cart (empty) to an anonymous one (with one entry). The point here is if we want to add or not cases for different scenarios like merge a customer cart (with one entry) to an anonymous one (with one entry) where the products   in each cart are different or where the products are the same. I think we need to agree in the team.

Reply:
@herzog31 the test was wrong anyway because it was depending on the previous one.  I refactor it and add it the case where the customer already has something in his cart.